test/parallel/test-http-client-headers-array.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+require('../common');
+
+const assert = require('assert');
+const http = require('http');
+
+function execute(options) {
+  http.createServer(function(req, res) {
+    const expectHeaders = {
+      'x-foo': 'boom',
+      cookie: 'a=1; b=2; c=3',
+      connection: 'close'
+    };
+
+    // no Host header when you set headers an array
+    if (!Array.isArray(options.headers)) {
+      expectHeaders.host = `localhost:${this.address().port}`;
+    }
+
+    // no Authorization header when you set headers an array
```

Comment:
`as` array?

And it would be nice to use caps for the first character of each comment.

Both applies to the comment above as well.